packages/react-scripts/fixtures/kitchensink/src/App.js

```diff
@@ -1,23 +1,22 @@
-import React from 'react';
+import React, { Component, Children, cloneElement } from 'react';
 
 class BuiltEmitter extends React.Component {
-  constructor(props) {
-    super(props)
-
-    this.callWhenDone = done => done();
+  componentDidMount() {
+    const only = Children.only(this.props.children);
+    if (!Component.isPrototypeOf(only)) {
+      this.notifyRendered();
+    }
   }
 
-  componentDidMount() {
-    this.callWhenDone(() => document.dispatchEvent(new Event('ReactFeatureDidMount')));
+  notifyRendered() {
+    document.dispatchEvent(new Event('ReactFeatureDidMount'));
   }
 
   render() {
-    const feature = React.cloneElement(React.Children.only(this.props.children), {
-      setCallWhenDone: done => {
-        this.callWhenDone = done;
-      }
+    const { children } = this.props;
+    const feature = cloneElement(Children.only(children), {
+      notifyRendered
     });
-
     return <div>{feature}</div>;
   }
 }
```

Comment:
Below, why do we call `setFeature(null)` if URL doesn't match? Why not throw?
We should make sure we don't mistype fixture name in the test. If feature doesn't exist we should fail.